tests/integration/test_refreshable_mat_view/test.py

```diff
@@ -0,0 +1,864 @@
+def parse_ch_datetime(date_str):
+    if date_str is None:
+        return None
+    return datetime.strptime(date_str, "%Y-%m-%d %H:%M:%S")
+
+
+# Add cases for:
+# Randomize bigger than interval
+# executed even with empty
+"AFTER 1 MINUTE RANDOMIZE FOR 3 YEAR",
+"EVERY 1 MINUTE OFFSET 1 SECOND RANDOMIZE FOR 3 YEAR",
+
+# special case
+"EVERY 999 MINUTE"
+
+# should be restricted:
+# Two different units in EVERY
+"EVERY 1 YEAR 0 MONTH",
+"EVERY 1 YEAR 1 MONTH",
+"EVERY 1 YEAR 2 MONTH",
+# "EVERY 1 YEAR 3 MONTH",
+# "EVERY 1 YEAR 4 MONTH",
+# "EVERY 1 YEAR 8 MONTH",
+# "EVERY 1 YEAR 9 MONTH",
+# "EVERY 1 YEAR 10 MONTH",
+# "EVERY 1 YEAR 11 MONTH",
+"EVERY 1 YEAR 12 MONTH",
+# "EVERY 1 YEAR 13 MONTH",
+# "EVERY 1 DAY",
+
+# should be restricted?
+"EVERY 1 YEAR 2 YEAR",
+"EVERY 1 MONTH 2 MONTH",
+"AFTER 1 YEAR 2 YEAR",
+"AFTER 1 MONTH 2 MONTH",
+
+"EVERY 1 DAY 0 HOUR",
+"EVERY 1 DAY 1 HOUR",
+"EVERY 1 DAY 2 HOUR",
+"EVERY 1 DAY 23 HOUR",
+"EVERY 1 DAY 24 HOUR",
+"EVERY 1 DAY 28 HOUR",
+# "EVERY 1 DAY 29 HOUR",
+# "EVERY 1 DAY 30 HOUR",
+# "EVERY 1 DAY 31 HOUR",
+"EVERY 1 DAY 32 HOUR",
+
+# Interval shouldn't contain both calendar units and clock units (e.g. months and days)
+# "EVERY 1 YEAR 10 MONTH 3 DAY 7 HOUR 5 MINUTE 30 SECOND",
+
+
+INTERVALS_EVERY = [
+    "EVERY 60 SECOND",
+    "EVERY 1 MINUTE",
+    "EVERY 1 HOUR",
+    "EVERY 1 DAY",
+    "EVERY 1 MONTH",
+    "EVERY 1 YEAR",
+    "EVERY 1 WEEK RANDOMIZE FOR 1 HOUR",
+    "EVERY 2 WEEK OFFSET 5 DAY 15 HOUR 10 MINUTE",
+    "EVERY 1 MONTH OFFSET 5 DAY 2 HOUR 30 MINUTE 15 SECOND",
+    "EVERY 1 MONTH OFFSET 1 DAY RANDOMIZE FOR 10 HOUR",
+    "EVERY 1 WEEK",
+    "EVERY 2 WEEK",
+    "EVERY 3 WEEK",
+    "EVERY 4 WEEK",
+    "EVERY 5 WEEK",
+]
+
+INTERVALS_AFTER = [
+    "AFTER 59 SECOND",
+    "AFTER 60 SECOND",
+    "AFTER 61 SECOND",
+    "AFTER 30 MINUTE",
+    "AFTER 9999 MINUTE",
+    "AFTER 2 HOUR",
+    "AFTER 2 DAY",
+    "AFTER 2 WEEK",
+    "AFTER 2 MONTH",
+    "AFTER 2 YEAR",
+    "AFTER 2 MONTH 3 DAY",
+    "AFTER 2 MONTH 3 DAY RANDOMIZE FOR 1 DAY",
+    "AFTER 1 YEAR RANDOMIZE FOR 11 MONTH",
+    "AFTER 1 MONTH",
+    "AFTER 1 MONTH 0 DAY",
+    "AFTER 1 MONTH 1 DAY",
+    "AFTER 1 MONTH 3 DAY",
+    "AFTER 1 MONTH 50 DAY",
+    "AFTER 1 YEAR 10 MONTH",
+    "AFTER 1 YEAR 10 MONTH 3 DAY",
+    "AFTER 1 YEAR 10 MONTH 3 DAY 7 HOUR",
+    "AFTER 1 YEAR 10 MONTH 3 DAY 7 HOUR 5 MINUTE 30 SECOND",
+]
+
+
+@pytest.fixture(scope="function")
+def rmv_schedule_teardown():
+    node.query("DROP TABLE IF EXISTS test_rmv")
+
+
+def expect_rows(rows, table="test_rmv"):
+    inserted_data = node.query_with_retry(
+        f"SELECT * FROM {table}",
+        parse=True,
+        check_callback=lambda x: len(x) == rows,
+        retry_count=100,
+    )
+    assert len(inserted_data) == rows
+
+
+@pytest.mark.parametrize("interval", INTERVALS_AFTER + INTERVALS_EVERY)
+@pytest.mark.parametrize(
+    "append",
+    [True, False],
+)
+@pytest.mark.parametrize(
+    "empty",
+    [True, False],
+)
```

Comment:
Slow. Consider:
 * Generating `append` at random instead having it as a parameter and doubling the test runtime. It shouldn't affect scheduling anyway. Similarly remove or randomize unimportant parameters in other tests.
 * Doing many (all?) intervals in one go. Because IIUC in CI each test usually runs separately and starts whole new clickhouse servers. So having 2 parameter combinations would be much faster than 70.

Reply:
The scheduling model has been removed with this test.